File: src/main/java/com/sothawo/springdataelastictest/foo/FooService.java
/*
 * (c) Copyright 2023 sothawo
 */
package com.sothawo.springdataelastictest.foo;

import org.springframework.data.domain.Sort;
import org.springframework.data.elasticsearch.core.SearchHit;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
@Service
public class FooService {
    private final FooRepository repository;

    public FooService(FooRepository repository) {
        this.repository = repository;
    }

    public Flux<SearchHit<Foo>> test() {
        var foo = new Foo();
        foo.setId("1");
        foo.setLongValue(42L);
        return repository.save(foo)
                .thenMany(repository.searchBy(Sort.by("longValue")));
    }
}

File: src/main/kotlin/com/sothawo/springdataelastictest/so/FooService.kt
package com.sothawo.springdataelastictest.so

import org.springframework.data.elasticsearch.client.elc.NativeQuery
import org.springframework.data.elasticsearch.core.ElasticsearchOperations
import org.springframework.data.elasticsearch.core.SearchHits
import org.springframework.stereotype.Service

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
@Service
class FooService(
		private val operations: ElasticsearchOperations
) {
		fun nativeQuery(): SearchHits<Foo> {
val query = NativeQuery.builder().withQuery { q1 ->
		q1.bool { b ->
				listOf(1L, 2L).forEach { t ->
						b.filter { q2 ->
								q2.term { tq ->
										tq.field("tags.id").value(t)
										tq
								}
						}
				}
				b
		}
}.build()
				return operations.search(query, Foo::class.java)
		}
}
